conf/nginx/security.conf.inc

```diff
@@ -25,18 +25,29 @@ ssl_dhparam /usr/share/yunohost/ffdhe2048.pem;
 # Follows the Web Security Directives from the Mozilla Dev Lab and the Mozilla Obervatory + Partners
 # https://wiki.mozilla.org/Security/Guidelines/Web_Security
 # https://observatory.mozilla.org/
-{% if experimental == "True" %}
-more_set_headers "Content-Security-Policy : upgrade-insecure-requests; default-src https: data: blob: ; object-src https: data: 'unsafe-inline'; style-src https: data: 'unsafe-inline' ; script-src https: data: 'unsafe-inline' 'unsafe-eval'";
-{% else %}
-more_set_headers "Content-Security-Policy : upgrade-insecure-requests";
-more_set_headers "Content-Security-Policy-Report-Only : default-src https: data: blob: ; object-src https: data: 'unsafe-inline'; style-src https: data: 'unsafe-inline' ; script-src https: data: 'unsafe-inline' 'unsafe-eval'";
-{% endif %}
 more_set_headers "X-Content-Type-Options : nosniff";
 more_set_headers "X-XSS-Protection : 1; mode=block";
 more_set_headers "X-Download-Options : noopen";
 more_set_headers "X-Permitted-Cross-Domain-Policies : none";
 more_set_headers "X-Frame-Options : SAMEORIGIN";
 
+# Here we add a generic CSP headers only if the app doesn't define it.
+# NB: if the <meta tag is used the most restrictive rules will be used
+{% if experimental == "True" %}
+header_filter_by_lua_block {
+  if ngx.header["Content-Security-Policy"] == nil or ngx.header["Content-Security-Policy"] == "" then
```

Comment:
I am not totally sure about `ngx.header["Content-Security-Policy"] == ""`

Reply:
I have tested it adds the field if the header is not set by the app, but i have not tested it doesn't if the header (from reverse proxy) exists...

Reply:
I have now test in the 2 use case. For me this patch is ok

Reply:
Imho this is tweak is still debatable because, as far as I understand, apps that *do* implement CSP may provide it via an HTML head tags, not via a CSP HTTP header (though i don't have any quantitative data on what's the most widespread practice). So this lua tweak still overrides legit CSP headers and therefore reduce the security in that case ...

Reply:
That's a good point - headers set by nGinX would also override http-equiv directives in the HTML.  And no security directives should be set by nGinX if any of the security directives are set by the app, either as headers or http-equiv directives.